be1-go/hub/standard_hub/mod.go

```diff
@@ -295,7 +266,7 @@ func (h *Hub) SendGreetServer(socket socket.Socket) error {
 
 	socket.Send(buf)
 
-	h.peersGreeted = append(h.peersGreeted, socket.ID())
+	h.peers.AddPeerGreeted(socket.ID())
```

Comment:
can the Send() fail ? if yes, is it still ok to AddPeerGreated() ?

Reply:
The Socket interface doesn't define any return value for it so the function Send() can't really fail by itself. 
It sends the message to the WritePump() goroutine which logs errors for calls to NextWriter() or Close() but, from what I see, it doesn't handle the errors following the call to Write(). I can open an issue for this so that it's dealt with in the future. There are many parts in the code where we just assume that Send() doesn't fail.

Reply:
yes please open an issue referencing this